### ISSUE TITLE
Elementor Pro 2.6.0 (Beta 4 Release)

### ISSUE THREAD

**Issue by shilo-ey** [elementor/elementor#8506]
===========
## [Beta 4 UPDATED - PLEASE READ BELOW!](https://github.com/elementor/elementor/issues/8506#issuecomment-513854923)
===========

Elementor v2.6 Pro beta is here and we would love to get feedback from you!
Liked the new Icon Library? You will LOOOVE what we have in Pro!

**Before you test, please make sure you are using:**

- **Elementor Core v2.6.0**
- **Elementor Pro v2.6.0 beta2**

## Meet Custom Icon Libraries (#110)

![3](https://user-images.githubusercontent.com/45009641/60905029-e7cb1000-a27c-11e9-91a7-fc38f3475c42.png)

One of our top priorities is making sure Elementor Pro users get the most advanced and cutting edge design capabilities. With this release, it's the first time users are able to integrate to three of the leading icon libraries font generators: Fontello, IcoMoon, and Fontastic.

This way, you can be sure you'll always be able to find just the right icon for your project.

All of your custom icon sets are accessible through a single interface, allowing you to create your favorite icons packs, and use them throughout your workflow.

The upload process is super simple, and once it is done, all the icons will be available inside the new Icon Library.

Every icon you use, no matter which library, is edited and styled using Elementor's native styling controls. It completely changes the usual way designers work with custom icons. Don’t settle for pre-made icon sets. Build your own sets and style them inside Elementor.
 
**Test flow:**

- Go to one of the supported icon libraries generators mentioned above and create an icon set (make sure to use unique prefix per icons set).
- In your dashboard go to Elementor > Custom Icons and click on “Add New” button.
- Set your title, and upload your pre-made Icon set.
- Make sure all of the details mentioned are correct and the icons are being displayed properly.
- Click “Publish” and create a new Page/Post.
- Add any widget uses an icon, and select your newly created library
- Browse through your uploaded icons, insert one and play with the design settings.
- Make sure everything works as expected in both the Editor and on Frontend.

## Font Awesome Pro Integration (#4430)
We added Font Awesome 5 for all our users, but for all you Pro users, you get a whole lot more, with easy integration to Font Awesome 5 Pro.

All you need to do is add your Font Awesome 5 Pro kit ID, and you'll get instant access to 5,300+ smashing Font Awesome Pro icons directly to your Elementor Icon Library.

**Note:** this feature is only available to Font Awesome Pro subscribers.

**Test flow:**

- In your dashboard go to Elementor > Settings > Integrations tab.
- Scroll to the Font Awesome Pro row, fill in your Font Awesome Pro Kit ID and click “Validate Kit ID”
- After receiving confirmation click on “Save Changes” button
- Create a new page and insert any widget uses an icon
- Browse throughout the new icon families added and insert one of the presented icons.
- Make sure everything works as expected in both the Editor and on Frontend.

## reCAPTCHA v3 integration (#8213, #6039, #7165)

With the recent Facebook security outage, it’s clear that website security is important for every site, large or small...

Luckily, we have your back and added reCAPTCHA v3 to our form user validation.

![reC](https://user-images.githubusercontent.com/45009641/61056850-c1cc7980-a3fc-11e9-8f55-8d21728e9e5e.png)

reCAPTCHA v3 works like magic. It monitors the user with almost no interruption and gives a quality score behind the scenes based on the user’s actions. This way, only suspicious users (AKA bots) are given the validation fields, and the experience of screened users remains uninterrupted.

As an admin, you can set the score threshold you desire, zero being the ‘loose’  score and one being the ‘strict’ score.

Much like FontAwesome 5 Pro, the integration is simple and straightforward. 

Better luck elsewhere, hackers. This user uses Elementor forms!

**Test flow:**

- In your dashboard go to Elementor > Settings > Integrations tab.
- Scroll to the Google reCAPTCHA v3 row, fill in your Site Key, Secret Key and the Score Threshold you are willing to accept as valid score for users.
- Click on “Save Changes” button.
- Create a new page and insert a form.
- Add new field and set it’s type to reCAPTCHA v3.
- Make sure everything works as expected on Frontend.

## Popup Exit Animation (#7063)

Many of our users wanted us to add the Exit Animation feature to our Popups, and finally it's here.

For exit animation, once the user closes the popup, instead of the popup disappearing immediately, you can pick a more subtle or unique exit like fade out.

**Test flow:**

- Go to Dashboard > Templates > Popups > Add New.
- Name your template and click ‘Create Template’.
- Choose a template from the Library, or built your popup from scratch.
- Under ‘Popup Settings’ set your Exit Animation
- Set different Exit Animation for mobile
- Set your triggers and conditions
- Make sure everything works as expected on Frontend.

## Support taxonomy ACF Dynamic Tags in Archive pages (#5147)

Archive pages are super important, whether if it’s the blogs front page, the different products categories or even in a hotel archive pages.

In this release, we took care of the dynamic structure of those pages. 
For example, if you have a shoe shop you will be able to place a dynamic banner that will introduce an image of men shoes from an ACF field, and women shoes in the women archive page. 

The options for customizing your taxonomy archive pages dynamically are limitless, starting from customizing tags, categories, authors or any other custom taxonomy you could think of.

Start creating your personalized archive pages with Elementor Pro v2.6 now.  

**Test flow:**

- Use ACF to add a custom text field and custom image gallery to your ‘Categories’ taxonomy.
- Go to Posts > Categories and fill in the fields specified above in one of your categories.
- In your dashboard go to Templates > Theme Builder and create an Archive template.
- In your archive template place a Heading widget and Image Gallery widget and set its source to Dynamic > ACF field > Set the ‘Key’ to the corresponding field.
- Make sure you see the values you added and that everything works as expected on Frontend.

----------------

## More Improvements

- Tweak: Added Dynamic capabilities for Form Redirect (#7552)
- Tweak: Added Logged In Message styling options for Login widget (#7928)
- Tweak: Added `none` breakpoint option to Nav-menu widget (#7916)
- Tweak: Added option to place post terms without links (#8366)
- Tweak: Added `elementor/query/query_results` hook to `query-control` to allow full control over results for developers (#7912)​
- Tweak: Price Table - allow choosing Heading level (#8090)​
- Tweak: Show popup on dynamic click even when `Avoid multiple popups` option is selected (#8189)
- Tweak: Added `Effects relative to` control to Motion Effects
- Tweak: Allow shortcodes in HTML field
- Tweak: removed donReach support from Share Buttons widget
- Tweak: Changed MailChimp `List` label to `Audience`
- Tweak: Improved Entrance & Exit animation behavior in Popup
- Tweak: Added `Deprecated Notice` control to `Archive products` & `Woo-products` widgets
- Tweak: Add "Recent Posts" Title for archive title dynamic tag in Theme Builder​
- Tweak: Added condition to show `Centered Slides` control in Media Carousel widget
- Tweak: Added notice bars in the Editor for Expired license and Activate license
- Tweak: Replaced `select` control with `choose` control in Price List widget
- Tweak: Removed Font Awesome 4 dependencies from the Editor
- Tweak: Added description to Terms control in Query Control
- Tweak: Minor styling tweaks in the Popup publish modal
- Tweak: Hide ordering form in Products widget on front-page.
- Deprecated: `DB/save_editor` - Remove usage of this method ([Deprecation post](https://developers.elementor.com/v2-6-0-planned-deprecations/))
- Deprecated: `DB::get_plain_editor` - Remove usage of this method ([Deprecation post](https://developers.elementor.com/v2-6-0-planned-deprecations/))

--------------------------------

## How to Install the Beta Version
If you still haven't enabled beta testing in Elementor, please follow these steps:

- Go to Elementor > Tools > Versions
- Enable the beta tester feature
- Go to Dashboard > Updates and update to the latest beta version.

This will get you set up to test the new Elementor Pro v2.6.0 Beta.
 
### Notes

For bugs and remarks, please reply with a comment on the specific issue here on GitHub.

**Please keep this related to the Current beta Only! Any other issue or feature request, feel free to open a new issue.**

Designated time for release: `July 23, 2019 (Estimated)`

Many thanks for your support and help!
Cheers,
The Elementor Team

**Comment by donwattz1459** [comment]
Custom icon packs are not showing up. I made mine with Fontello. 

![Screen Shot 2019-07-10 at 12 22 27 AM](https://user-images.githubusercontent.com/16137152/60940279-e1528d80-a2a8-11e9-8edb-06ccbb8c684a.png)

**Comment by wtassis** [comment]
I have the same problem.

**Comment by KiQdev** [comment]
When having Pro beta 2 installed, I can no longer use SVG files like Undraw SVG because it says “undraw_houses3_xwf7.svg” has failed to upload.
Sorry, this file type is not permitted for security reasons.
So no longer possible to use SVG as before?

**Comment by shilo-ey** [comment]
Hi @KiQdev 

Since Elementor v2.6 we added SVG sanitizer to the locations Elementor allows you to upload SVG files, which in some cases those files may pose a security risk.

Because we do not interfere with WordPress's media library, we can't allow uploading those files without sanitizing the SVG files being uploaded.

Due to this reason, from now on you can upload SVG files to:
- Elementor SVG library (any widget with icon control > Upload SVG)
- Elementor Custom Fonts

If you wish to upload in the WP media library - please use 3rd party plugin or custom code to allow this option.

Thanks!

**Comment by KiQdev** [comment]
Ok. Yes I want to use as background in some cases.
Strange, just checked another site where I have pro beta 2, there it was
possible to upload to media library.

Den tors 11 juli 2019 17:26shilo-ey <notifications@github.com> skrev:

**Comment by ktse312** [comment]
Hello, after enable Beta Tester and clearing all cache and refreshing but still do not see option to update to Elementor Pro v2.6.0 Beta.

**Comment by kiudex** [comment]
With Font Awesome Pro activated, the icon library causes the tab to freeze every time I try to open ithe icon library through a widget. I tried to change an icon of the social icon widget and icon widget. 
<img width="1244" alt="Screen Shot 2019-07-12 at 16 04 23" src="https://user-images.githubusercontent.com/14236672/61112973-ad04ea00-a4bf-11e9-8984-e11a1f474258.png">

As soon as I click "Icon Library" the CPU usage goes up and the tab freezes.

**Comment by Lorenzoup** [comment]
After the update, the incoming animations no longer work in Google Chrome.

**Comment by inocampo** [comment]
Hello, thank you

In the widget Share Buttons not show the icon in the button print. I have Awesome Font PRO
![elementor issue](https://user-images.githubusercontent.com/1313634/61174760-35d86000-a56a-11e9-9702-35e61d80c8b9.png)

**Comment by shilo-ey** [comment]
We just released a new beta version, please update your version to Pro v2.6.0 Beta 3.

This beta release includes:

- Tweak: UI modifications to Custom Icons screens
- Tweak: Added support for 'Hide Title' in Theme Builder
- Fix: Copy-Paste style not pasting the Pointer option in Nav Menu widget (#8497)
- Fix: Missing Print icon in Share Buttons (#8506)

**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

Thanks again to everyone.

**Comment by shilo-ey** [comment]
[**@kiudex**](https://github.com/kiudex) commented on [Jul 12, 2019, 11:14 AM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-510795414 "2019-07-12T08:14:58Z - Replied by Github Reply Comments"):
> With Font Awesome Pro activated, the icon library causes the tab to freeze every time I try to open ithe icon library through a widget. I tried to change an icon of the social icon widget and icon widget. 
> As soon as I click "Icon Library" the CPU usage goes up and the tab freezes.


Can you please provide your system info?




[**@Lorenzoup**](https://github.com/Lorenzoup) commented on [Jul 12, 2019, 12:03 PM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-510811191 "2019-07-12T09:03:07Z - Replied by Github Reply Comments"):
> After the update, the incoming animations no longer work in Google Chrome.

Please contact our support at support @elementor.com for help us identifying in resolving this issue and mention that you were directed by me and reference this issue.


[**@inocampo**](https://github.com/inocampo) commented on [Jul 13, 2019, 8:33 PM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-511140125 "2019-07-13T17:33:34Z - Replied by Github Reply Comments"):
> Hello, thank you
> 
> In the widget Share Buttons not show the icon in the button print. I have Awesome Font PRO


Thanks, this issue is resolved in `beta3`, feel free to upgrade.


Thank you all in advance

**Comment by inocampo** [comment]
> Hello, thank you
> 
> In the widget Share Buttons not show the icon in the button print. I have Awesome Font PRO
> ![elementor issue](https://user-images.githubusercontent.com/1313634/61174760-35d86000-a56a-11e9-9702-35e61d80c8b9.png)

Solved in beta3

**Comment by kiudex** [comment]
@shilo-ey Here the requested system info for the environment having the issue:

[system-info-audio.kiudex.de-15-07-2019.txt](https://github.com/elementor/elementor/files/3392940/system-info-audio.kiudex.de-15-07-2019.txt)

**Comment by inocampo** [comment]
![error elementor](https://user-images.githubusercontent.com/1313634/61246307-25fc7f80-a714-11e9-9706-adacf8d7bdfa.png)
I have Beta3 and the icon size have a problem, please see the attached image

**Comment by ktse312** [comment]
I have Beta3. Icon not work with "Icon List" widget.

<img width="527" alt="icon list" src="https://user-images.githubusercontent.com/52795498/61274912-565d2000-a7df-11e9-8ad2-340d14788aec.png">

**Comment by inocampo** [comment]
Hello,
I have `Beta3` and I went up Fontello .zip but in the image we can see that `Elementor` does not show the icons and the gallery name.

The .zip of `Fontello` has 148 icons but `Elementor` shows 165 icons, I did download 165 svg in `Flat Icon`, but in Fontello only 148.

![fontello](https://user-images.githubusercontent.com/1313634/61293244-ddd37080-a798-11e9-8326-b5cb0be1f14e.png)

**Comment by inocampo** [comment]
I have `Beta3`, this is the SVG icon in button widget, something icons its ok, but other icons with errors, I download SVG icon in `Flat Icons`. 

![Error SVG Icon](https://user-images.githubusercontent.com/1313634/61300215-2f82f780-a7a7-11e9-8773-51f0d0e1519b.png)

**Comment by scottyg7** [comment]
I've just notices with beta3 (may have been beta2) that my Posts Widget looks to be adding blank space equal to the height of the post image: 
![image](https://user-images.githubusercontent.com/6504194/61323178-50663f80-a7dd-11e9-9ec2-b63ca0e74db3.png)

**Comment by shilo-ey** [comment]
Hey @kiudex 
From your system info we can see that you have some JS conflicts. Please contact our support at support @elementor.com for guidance in resolving this issue and mention that you were directed by me and reference this issue.


[**@inocampo**](https://github.com/inocampo) commented on [Jul 15, 2019, 11:28 PM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-511557242 "2019-07-15T20:28:41Z - Replied by Github Reply Comments"):
> I have Beta3 and the icon size have a problem, please see the attached image

In which widget? can you attach a link to a page that experiencing this issue?


[**@ktse312**](https://github.com/ktse312) commented on [Jul 16, 2019, 10:36 AM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-511702356 "2019-07-16T07:36:06Z - Replied by Github Reply Comments"):
> I have Beta3. Icon not work with "Icon List" widget.
> 

Please check your console for 404 errors. 
Make sure your shim.js is loaded by going to `Dashboard > Elementor > Settings > Advanced > Load Font Awesome 4 Support`	and setting the option to `Yes`.
Anyway - this is not related to the Pro beta. Please contact our support if this issue persists.





[**@inocampo**](https://github.com/inocampo) commented on [Jul 16, 2019, 3:19 PM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-511792161 "2019-07-16T12:19:55Z - Replied by Github Reply Comments"):
> Hello,
> I have `Beta3` and I went up Fontello .zip but in the image we can see that `Elementor` does not show the icons and the gallery name.
> 
> The .zip of `Fontello` has 148 icons but `Elementor` shows 165 icons, I did download 165 svg in `Flat Icon`, but in Fontello only 148.


Please send us the zip file generated from Fontello.

[**@inocampo**](https://github.com/inocampo) commented on [Jul 16, 2019, 4:56 PM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-511826866 "2019-07-16T13:56:15Z - Replied by Github Reply Comments"):
> I have `Beta3`, this is the SVG icon in button widget, something icons its ok, but other icons with errors, I download SVG icon in `Flat Icons`. 
> 

Make sure you don't have any inline settings inside your `svg` files.



[**@scottyg7**](https://github.com/scottyg7) commented on [Jul 16, 2019, 10:21 PM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-511950724 "2019-07-16T19:21:11Z - Replied by Github Reply Comments"):
> I've just notices with beta3 (may have been beta2) that my Posts Widget looks to be adding blank space equal to the height of the post image: 
> 
> In edit mode, it looks fine...just the published post is showing the extra space.

Can you please attach a link to a page that experiencing this issue?



Thanks!

**Comment by inocampo** [comment]
@inocampo 2. This is the `Fontello` .zip that I'm using
[fontello-caa3e16a.zip](https://github.com/elementor/elementor/files/3401723/fontello-caa3e16a.zip)

@inocampo 3. ¿How can I check the `svg` inline setting?

**Comment by tristanleboss** [comment]
@shilo-ey  Can you add this: #8176

It's a one line improvement with no performance impact and no need to test.

**Comment by wtassis** [comment]
> 
> 
> Custom icon packs are not showing up. I made mine with Fontello.
> 
> ![Screen Shot 2019-07-10 at 12 22 27 AM](https://user-images.githubusercontent.com/16137152/60940279-e1528d80-a2a8-11e9-8edb-06ccbb8c684a.png)

After upgrade to beta3 still having the same issue.

**Comment by lucafagherazzi** [comment]
Hi, I have Elementor 2.6.5 and Elementor Pro 2.6.0-beta3, I encountered two problems with the template widget.
1 when I insert a new template widget it finds no result;
2 on a template widget created a couple of months ago it cancels the style customizations every time I open the post with Elementor.

<img width="292" alt="Schermata 2019-07-19 alle 19 42 39" src="https://user-images.githubusercontent.com/40598133/61555691-4011c700-aa60-11e9-9582-9c164d2df6c8.png">
<img width="642" alt="Schermata 2019-07-19 alle 19 47 19" src="https://user-images.githubusercontent.com/40598133/61555707-4a33c580-aa60-11e9-82f6-c861dd66d950.png">
<img width="1446" alt="Schermata 2019-07-19 alle 19 43 50" src="https://user-images.githubusercontent.com/40598133/61555733-59b30e80-aa60-11e9-8755-8a6616aadaad.png">
<img width="985" alt="Schermata 2019-07-19 alle 19 45 41" src="https://user-images.githubusercontent.com/40598133/61555747-60418600-aa60-11e9-9f2f-f46fa7706503.png">

**Comment by dudaster** [comment]
With the latest update Elementor Custom Skin doesn't show ACF fields anymore, it does work with Pods though. Is there something I can do on my end?

**Comment by dudaster** [comment]
this line from acf-text.php  messes with the ability  to use ACF fields in a loop: 

`44. $field = get_field_object( $field_key, get_queried_object() );`

Is there a way to bypass that?

**Comment by dudaster** [comment]
NVM I found a way.

**Comment by shilo-ey** [comment]
We just released a new beta version, please update your version to Pro v2.6.0 Beta 4.

This beta release includes:

- Tweak: Use JS instead of JSON files
- Tweak: Added style controls for HTML field in Form widget
- Tweak: Custom icons UI tweaks
- Fix: Form widget Date picker makes the Popup builder disappear (#7240)
- Fix: Sticky element stop point stops working on viewport resize (#7884)

**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

Thanks again to everyone.

**Comment by shilo-ey** [comment]
[**@inocampo**](https://github.com/inocampo) commented on [Jul 17, 2019, 3:10 PM GMT+3](https://github.com/elementor/elementor/issues/8506#issuecomment-512227088 "2019-07-17T12:10:20Z - Replied by Github Reply Comments"):
> @inocampo 2. This is the `Fontello` .zip that I'm using
> [fontello-caa3e16a.zip](https://github.com/elementor/elementor/files/3401723/fontello-caa3e16a.zip)
> 
> @inocampo 3. ¿How can I check the `svg` inline setting?


Hi @inocampo 

Please note that in your `config.json` generated file there are 165 items, this is why Elementor Pro recognizes 165 icons.

As for why it's not appearing in your Custom Icons screen - it's due to the fact that there's no name specified in Fontello prior to downloading the generated zip file.


![image](https://user-images.githubusercontent.com/45009641/61702039-ccb1d300-ad47-11e9-88f8-fdc46bd3a2c6.png)


For example, here's an import of your 'config.json' after we added a name:
[fontello-f1d5d5a0.zip](https://github.com/elementor/elementor/files/3421371/fontello-f1d5d5a0.zip)

Thanks

**Comment by shilo-ey** [comment]
Elementor Pro v2.6.0 is out.

Thank you all for helping out!

Feel free to update.

**Comment by eric-cathell** [comment]
Elementor Pro 2.6.1 breaks OceanWP Social Sharing Widgets

Deactivated
![image](https://user-images.githubusercontent.com/53321934/61918816-d5113600-af20-11e9-97bf-af86b88b9a24.png)

![image](https://user-images.githubusercontent.com/53321934/61918831-e1958e80-af20-11e9-8c0b-639ce317d5ab.png)

Activated
![image](https://user-images.githubusercontent.com/53321934/61918853-f7a34f00-af20-11e9-9945-c42acf9b1f36.png)
![image](https://user-images.githubusercontent.com/53321934/61918863-02f67a80-af21-11e9-93b4-edc9b8ab17f6.png)

**Comment by tristanleboss** [comment]
@eric-cathell  You should open a new issue.
